nixos/modules/config/update-users-groups.pl

```diff
@@ -235,7 +235,10 @@ sub parseUser {
     # Ensure home directory incl. ownership and permissions.
     if ($u->{createHome} and !$is_dry) {
         make_path($u->{home}, { mode => oct($u->{homeMode}) }) if ! -e $u->{home};
-        chown $u->{uid}, $u->{gid}, $u->{home};
+        system("chown",
```

Comment:
This chown runs on every activation. Is this not super slow for a decent sized home?

Reply:
Probably, I only tested it on new installations. It would also be fine for me if this only runs on the first system Boot for my use case, but that would probably add quite a bit of complexity and some state with checking if a specific file exists or something similar.

Reply:
How about the following alternative instead of adding more code to this perl script.
In your own configuration you can add the following line to `systemd.tmpfiles.rules`


```
Z     /your/home 700 youruser yourgroup -           -
```

It will achieve the same thing that this option would do.
systemd-tmpfiles is always run after the activation phase. That means the user is ensured to be present. However I have somehow doubts you always want to run this recursive chown, but maybe only once after some event happened?
I think for your use case of fixing up permissions after creating a disk image,
it might be better to some activation script that would do something like that:

```
if [ ! -f /var/lib/nixos/first-boot ]; then
  chown -R /your/home youruser:yourgroup
  touch /var/lib/nixos/first-boot
fi
```

The other alternative would be to pre-allocate a fixed uid/gid, so that during installation the permissions will be already correct.

Reply:
> Probably, I only tested it on new installations. It would also be fine for me if this only runs on the first system Boot for my use case, but that would probably add quite a bit of complexity and some state with checking if a specific file exists or something similar.

I don't think my code above is that much complexity. Otherwise you can still use the systemd-tmpfiles rules option, which is just one line of configuration.

Reply:
Yeah fair enough doing something like you wrote above and putting it in system.activationScripts is probably the better solution. I will close this PR.